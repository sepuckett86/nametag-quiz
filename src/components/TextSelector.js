import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ handleChange, text }) {
  return (
    <label>
      Text: <input name="text" type="text" value={text} onChange={handleChange}/>
    </label>
  );
}

TextSelector.propTypes = {
  handleChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
};

export default TextSelector;
