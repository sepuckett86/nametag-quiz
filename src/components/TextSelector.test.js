import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('renders TextSelector', () => {
    const wrapper = shallow(
      <TextSelector 
        text="Some text here"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
