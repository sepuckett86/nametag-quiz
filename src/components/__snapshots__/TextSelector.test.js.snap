// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders TextSelector 1`] = `
<label>
  Text: 
  <input
    name="text"
    onChange={[Function]}
    type="text"
    value="Some text here"
  />
</label>
`;
